docs/codestyle.md

```diff
@@ -0,0 +1,58 @@
+# WikiTree code style
+
+## Code style
+
+we aim for consistency, generality, readability and reduction of git diffs. Similar language constructs are formatted with similar rules. Style configuration options are deliberately limited and rarely added. Previous formatting is taken into account as little as possible.
+
+## Basic rules
+
+| Rule | Value | Note |
+| ---  | :---: | --- |
+| Line length | 120 characters |  |
+| Indentation | 4 spaces |  |
```

Comment:
This is my personal preference - I'm python coder and this is golden standard in python. I've thought about 2 spaces (also suggested by [Google JS guidelines](https://google.github.io/styleguide/jsguide.html#formatting-block-indentation)) and it's more compact, but I'm used to 4 spaces and those 2 doesn't look so clear and it's IMHO harder to orient, but thats what I feel.

I'm against using tabs 
- editors can be set to make spaces on tab key press
- Github makes ver large spaces for tab
- it doesn't really matter it you save some characters (1 character for tab vs N characters for every space)

Reply:
I agree. Two spaces is not enough, and we should definitely use four. I also agree they should be actual spaces. Editors can just do the replacement so the tab key itself works as expected. But storing tab characters in the file means everything gets out of alignment when viewed later.

The .prettierrc file that's currently in the repo sets tabWidth to two (2) spaces for HTML and CSS files. Is there a reason not to just have those files also use four-space indentation?  (It also has an "override" for .js files, but that sets the tabWidth to the default 4; that's probably unnecessary.)